ddl/generated_column.go

```diff
@@ -293,6 +296,11 @@ func (c *illegalFunctionChecker) Enter(inNode ast.Node) (outNode ast.Node, skipC
 		if !isFuncGA {
 			c.hasNotGAFunc4ExprIdx = true
 		}
+	case *driver.ValueExpr:
+		if node.Datum.Kind() == types.KindNull {
```

Comment:
It would forbid this case:
``` sql
mysql> CREATE TABLE tsg(a INT, b INT, INDEX c((concat(NULL, 3))));
Query OK, 0 rows affected (0.02 sec)
```

Reply:
@7yyo a gentle ping

Reply:
@wjhuang2016 Not all functions allow `NULL`, 
```mysql
mysql> select @@tidb_allow_function_for_expression_index;
+--------------------------------------------+
| @@tidb_allow_function_for_expression_index |
+--------------------------------------------+
| lower, md5, reverse, upper, vitess_hash    |
+--------------------------------------------+
1 row in set (0.00 sec)
```
I refer to mysql to distinguish functions that are already GA.

Reply:
@xhebox cc

Reply:
It makes sense for me. I did not think much more than the original issue. You could re-request review from wjhuang2016.

Reply:
After communicating with @wjhuang2016 , close this PR, this may be a compatibility issue.